src/controls/combinators/CombinatorSchemaSwitcher.tsx

```diff
@@ -36,16 +36,13 @@ export function CombinatorSchemaSwitcher({
     uischema.options as Record<string, unknown>,
   ) as OneOfControlOptions
   const oneOfOptionType = appliedUiSchemaOptions.optionType
-  const prevSelectedIndex = usePreviousValue(selectedIndex)
+  const prevSelectedIndex = usePreviousIndex(selectedIndex)
   const [dataForPreviousSchemas, setDataForPreviousSchemas] = useState<
     Record<number, unknown>
   >({})
 
   useEffect(() => {
-    if (
```

Comment:
Looks like this useEffect is doing a lot, normally you'd want to break the this into multiple if the side effects are not exactly correlated, otherwise you could run into unexpected behaviors.

Reply:
I was able to move a chunk of the logic into `onChange`. Let me know what you think 😄

Reply:
The `useEffect` is now gone.